src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -202,7 +206,7 @@ private async Task CheckContainedHomeRouteAsync(string containerContentItemId, C
 
                 foreach (var jItem in jItems.Cast<JsonObject>())
                 {
-                    var contentItem = jItem.ToObject<ContentItem>();
+                    var contentItem = jItem.ToObject<ContentItem>(_jsonSerializerOptions);
```

Comment:
Doesn't `ContentItem` have a custom converter, and a private `JsonObject`? This would then only be the Part/Field conversion to types that would need this serializer. If that's the case then all `ToObject<ContentItem>` don't need it.

Reply:
I don't think `ContentItem` as a converted. and event if it does, then that converted would be added to the `JsonSerializationOption` so we would have to provide to any place that use `ContentItem` so it is properly serialized.

Reply:
![image](https://github.com/OrchardCMS/OrchardCore/assets/1165805/edf2cffd-223c-4cb3-b4e4-ab0edd3d15f0)

And if you look at the code it merges the `Data` property without losing anything from it (there is no type resolution involved)

Reply:
Oh I see. So do you agree that we should not need to pass _jsonSerlizer when we are JObject.FromObject(ContentItem) or .ToObject<ContentItem>()?

Reply:
Yes that's what I am thinking. I am just sad that you had to find all these already in this PR.

Reply:
Yeah lots of changes. It's okay, it my fault I did not see it has a converter. I'll update this again when I have some more time